### DISCUSSION TITLE
Decide whether the claim ETH fees should be charged for all existing and future campaigns or only for the upcoming ones

### DISCUSSION THREAD

**Discussion by andreivladbrg** [sablier-labs/evm-monorepo#1039] [Polls]
In [this issue](https://github.com/sablier-labs/v2-core/issues/1032), the question is raised about whether we should charge the claim fee for all existing campaigns or only for campaigns created after the fee is activated.

**Scenario example:** The fee is initially set to 0, and campaign X1 is created. Half of the recipients claim their airdrop with zero fees. In the meantime, the fee is updated to, let’s say `0.00001 ETH`, and campaign X2 is created. All the recipients in X2 will pay the claim fee, but what about the remaining recipients in X1?

For more details, please refer to the comments in issue #1032.

@sablier-labs/everybody  

The question is:

**Comment by maxdesalle** [comment]
It should only be applied to newly created campaigns. Reason being that for a large part, existing users have used us because we were free. If we now suddenly charge, that will create a backlash.

And the number of people who are due to claim on existing campaigns is so small compared to the millions, billions of people who will claim in future campaigns, it's not worth it to hurt our brand for this.

**Comment by smol-ninja** [comment]
And another question is whether we should be allowed to update the claim fee on already deployed campaigns. Let's say, a creator launches a campaign when the claim fee was 0.001 ETH. After the campaign is launched, should we be able to update the claim fee for that campaign?﻿

**Comment by smol-ninja** [comment]
To @andreivladbrg, replying to [your comment](https://github.com/sablier-labs/v2-core/issues/1032#issuecomment-2340512786):

> We can hardcode a constant MAX_FEE_IN_WEI = 0.01e18 // 0.01 ETH

The native tokens on non-ETH EVMs can be highly volatile. Thus, hard coding it to anything can limit us to how much we can charge when the price of native token is very low. Whereas the same arguement does not apply to the protocol fee since the fee is in percentage.

Wdyt?

**Comment by IaroslavMazur** [comment]
Because of the bitter feeling of having someone modify (read: "worsen") the terms of the contract you've signed with them - and, especially, when this happens without your explicit acknowledgement and/or approval, I think it'd be better to not change the rules during the game.

**_However_**, if we properly educate/warn the Airstream campaign creators - at/before the campaign creation time - about the `claimFee` variable and that its value may change in the future (we should also give them a maximum possible value for the fee, imo), then, this is the way to go: easier for us to implement, while still being fair and transparent wrt the Airstream creators.

**Reply by PaulRBerg** [reply]
Educating users is always an additional burden and many will ignore the educational content.

Let's choose the path of least resistance and make the fee immutable.

**Comment by andreivladbrg** [comment]
**vox populi** 

6 votes, and there is a clear majority for the second option, closing
